lib/pub_sub/message.rb

```diff
@@ -11,20 +11,22 @@ def initialize(payload)
     end
 
     def process
-      validate_message!
+      begin
+        validate_message!
+      rescue PubSub::ServiceUnknown => e
+        return
+      end
       handler.process(data)
     end
 
     def validate_message!
       messages = PubSub.config.subscriptions[sender]
-      if messages.nil?
-        warning = "WARN: We received a message from #{sender} but we do " \
-                'not subscribe to that service.'
-        PubSub.logger.warn(warning)
-      end
-      unless messages.include?(type)
-        error = "We received a message from #{sender} but it was " \
-                "of unknown type #{type}."
+      if messages.blank?
```

Comment:
`blank` is a rails specific method, yet in [pub_sub.rb](https://github.com/westfieldlabs/pub_sub/blob/2002d999a6f541a2542501a377d37647bf2bb4ff/lib/pub_sub.rb#L8), there's a conditional based upon whether rails is available or not.  I'd be wary about tying this to rails just in case the authors of the gem intended it to be generic enough to be used in standalone ruby scripts with no rails support.